src/data_loader.py

```diff
@@ -0,0 +1,160 @@
+import os
+import PIL
+import torch
+import random
+import collections
+import numpy as np
+import pandas as pd
+
+
+
+class ImageSegmentaionDataset():
+    """Image Segmentation dataset"""
+    
+    def __init__(self, image_dict, loader_type, replacement=True, transform=None):
+        """
+        Args:
+        image_dict (dict): a dict of the following format:
+                        {"ships": {"imageId": "segmentation pixels"}， “w/ships”: ["imageId"]}
+        loader_type: (string)"classification" or "segmentation"
+        replacement: (bool) sample with/without replacement
+        transform (callable, optional): Optional transform to be applied on a sample.
+        """
+        self.image_dict = image_dict
+        self.loader_type = loader_type
+        self.replacement = replacement
+        # images_list is the name 
+        self.images_list = list(image_dict["ships"].keys()) + image_dict["w/ships"]
+        random.shuffle(self.images_list) 
+        self.transform = transform
+    
+    def __len__(self):
+        return len(self.images_list)
+    
+    def rle_decode(self, mask_rle, shape=(768, 768)):
```

Comment:
It makes sense to take this function out to a separate class, say: RLEHandler. The reason for it is because we could possibly use this function beyond the scope of this class. Additionally, (later) we'll need to implement rle_encode to prepare submission file. So all this functionality should be in one place

Reply:
Done, write the RLEHandler class in the file rle_handler.py